src/RealTimeSubscription.js

```diff
@@ -25,6 +29,25 @@ export default class RealTimeSubscription extends AbstractSubscription {
         }
     }
 
+    // passing publisherId separately to ensure it is lowercase (See call of this function in AbstractSubscription.js)
+    _decryptOrRequestGroupKey(msg, publisherId) {
+        try {
+            const newGroupKey = EncryptionUtil.decryptStreamMessage(msg, this.groupKeys[publisherId])
+            delete this.alreadyFailedToDecrypt[publisherId]
+            if (newGroupKey) {
+                this.groupKeys[publisherId] = newGroupKey
+            }
+            return true
+        } catch (e) {
```

Comment:
Best practice to only put try/catch only around code which you're expecting to fail. 

```js
let newGroupKey
try {
  newGroupKey = EncryptionUtil.decryptStreamMessage(msg, this.groupKeys[publisherId])
} catch (e) {
  // …
}
// happy path
delete this.alreadyFailedToDecrypt[publisherId]
// etc
```